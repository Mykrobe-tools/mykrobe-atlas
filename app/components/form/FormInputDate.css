.wrap {
  composes: wrap from "css/form.css";
}

.wrap + .wrap {
  margin-top: 8px;
}

.label {
  composes: label from "css/form.css";
}

.input {
  composes: input from "css/form.css";
}
