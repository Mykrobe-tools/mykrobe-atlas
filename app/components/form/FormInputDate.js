/* @flow */

import React, { Component, PropTypes } from 'react';
import moment from 'moment';
import DatePicker from 'react-datepicker';

import FormLabel from './FormLabel';

import styles from './FormInputDate.css';

class FormInputDate extends Component {

  render() {
    const {name, onChange, title, value} = this.props;
    return (
      <div className={styles.wrap}>
        <div className={styles.label}>
          <FormLabel
            htmlFor={name}
            label={title} />
        </div>
        <DatePicker
          className={styles.input}
          name={name}
          selected={value && value.length > 0 ? moment(value) : moment()}
          onChange={(date) => onChange(date)}
        />
      </div>
    );
  }
}

FormInputDate.propTypes = {
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  value: PropTypes.string
};

export default FormInputDate;
