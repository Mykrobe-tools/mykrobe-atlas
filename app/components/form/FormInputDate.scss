.wrap {
	composes: wrap from 'styles/form.scss';
}

.wrap + .wrap {
	margin-top: 8px;
}

.label {
	composes: label from 'styles/form.scss';
}

.input {
	composes: input from 'styles/form.scss';
}
