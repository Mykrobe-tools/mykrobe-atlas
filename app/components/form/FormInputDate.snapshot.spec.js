import React from 'react';
import renderer from 'react-test-renderer';
import FormInputDate from './FormInputDate';

jest.mock('react-datepicker', () => { return '';});

describe('FormInputDate component snapshot', () => {
  const name = 'loremIpsum';
  const title = 'Lorem Ipsum';
  const value = '2016-01-01';
  const onChange = jest.fn();

  it('renders correctly', () => {
    const component = renderer.create(
      <FormInputDate
        name={name}
        title={title}
        value={value}
        onChange={onChange} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
