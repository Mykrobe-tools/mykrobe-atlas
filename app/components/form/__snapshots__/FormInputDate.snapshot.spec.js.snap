exports[`FormInputDate component snapshot renders correctly 1`] = `
<div
  className="wrap">
  <div
    className="label">
    <label
      className="label"
      htmlFor="loremIpsum">
      Lorem Ipsum
    </label>
  </div>
  <
    className="input"
    name="loremIpsum"
    onChange={[Function]}
    selected={"2016-01-01T00:00:00.000Z"} />
</div>
`;
